neo/rawio/elanrawio.py

```diff
@@ -31,13 +31,21 @@ class ElanRawIO(BaseRawIO):
     extensions = ['eeg']
     rawmode = 'one-file'
 
-    def __init__(self, filename=''):
+    def __init__(self, filename=None, entfile=None, posfile=None):
         BaseRawIO.__init__(self)
         self.filename = filename
 
+        # check whether ent and pos files are defined
+        if not entfile:
```

Comment:
I think it would be better to compare the object identity via `is` here, to avoid puzzling  behavior in rare cases, e.g. when dealing with empty strings and pathlib objects.

Reply:
Yes, you're right, I also had issues with `is`. I can replace by `isinstance(entfile, string)` if you prefer

Reply:
Sorry, I think I was not clear enough in the first attempt. What I want to propose is something like this
```suggestion
        if not (entfile is None):
```
That is exactly only evaluated True if the default value is kept and not in any other case (e.g. empty list, string, ...)

Reply:
I think the `is` is going to produce a `SyntaxWarning` in Python >3.8 (see [here](https://docs.python.org/3.8/whatsnew/3.8.html#changes-in-python-behavior)). In the past I also had issues with the `is` when combined with inputs to scripts (e.g with ArgumentParser). So I would personally replace with `isinstance`

Reply:
I think the `is None` or `is not None` will still be valid for long time. This syntax wrning will is for other types no ?

Reply:
Yes probably. And yes, I think for the moment it's limited to strings and numbers. That said, I had issues with scripts inputs like `python myscript.py --entfile='myfile.ent'`. But probably as you both said, it's not the case here as it's a comparison with a `None`. I let you decide, let me know what fit best with the rest of the package

Reply:
Ah interesting. I didn't see this warning yet. Ok, but we only want to cover the case when `entfile` realliy is None, so would you then do `isinstance(entfile, type(None))`?

Reply:
@JuliaSprenger actually I would do `if not isinstance(entfile, str)`

Reply:
The change in Python 3.8 is not related to the situation here. `if not entfile` seems to me like the best  approach, as it handles both None and empty strings. If you want to be really strict, you could check if the file actually exists.

Reply:
(if you want to allow empty strings to pass, then `if entfile is not None`

Reply:
If you force your path to be of type string then you are excluding any approach of intelligent path handling, e.g. via [pathlib](https://docs.python.org/3/library/pathlib.html). For now I think @samuelgarcia is right and `is None` should be valid for long enough for us.

Reply:
Ah ok, didn't know pathlib. Then `entfile is None` is fine to me. I'll do the changes then. Thank you both !

Reply:
A downvote for `entfile is None`, because it means things will break further on if someone puts `entfile=""`, which is the sort of thing that happens when using Neo in scripts and applications. Probably the safest approach is to add a check that the file actually exists.

Reply:
@apdavison good point, I can still move to `os.path.isfile`. I thought it was performed somewhere else, internally, didn't check that

Reply:
@apdavison I think in the next steps the IO should at any point check the existence of the required files. However, the `is None` question is about how you deal with default values and when to replace these. I think the `if not filename` is not the ideal solution since it can lead to unexpected behavior. E.g. when the users explicitly don't want to load the file, not in the default location, but also not in the custom location, they might just set the filename to an empty string. With the `if not filename` construct this would still load the file from the default location, even though the default value was explicitely changed by the user.

Reply:
Sorry, this was probably getting too philosophical. I am ok with any of the 2 implementations, then it's up to @samuelgarcia to merge ;)

Reply:
I created https://github.com/NeuralEnsemble/python-neo/issues/938 so we can discuss this general issue independent of this PR.
@apdavison @samuelgarcia Would it be ok for you to merge the current version and potentially update this later depending on https://github.com/NeuralEnsemble/python-neo/issues/938 ?

Reply:
yes, sure

Reply:
yep. Lets merge